src/expression.js

```diff
@@ -206,7 +206,7 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprAtom(refShorthandDefaultPos)
   if (refShorthandDefaultPos && refShorthandDefaultPos.start) return expr
-  return this.parseSubscripts(expr, startPos, startLoc)
+  return this.parseSubscripts(expr, startPos, startLoc, expr.type === "ArrowFunctionExpression" && this.parenthesesCount === 0)
```

Comment:
For reference, this was the condition I tried:

```
expr.type === "ArrowFunctionExpression" && this.curContext() !== ct.p_expr
```

I'm not sure if it's a bug, but the current context is always `p_expr` in this position when dealing with arrow functions.